cabal-install/src/Distribution/Client/CmdPath.hs

```diff
@@ -0,0 +1,404 @@
+          <> "  "
+          <> pname
+          <> " path -z --output-format=key-value --installdir\n"
+          <> "    Print the installation directory, without taking project information into account.\n"
+    , commandUsage = \pname ->
+        "Usage: " <> pname <> " path [FLAGS]\n"
+    , commandDefaultFlags = defaultNixStyleFlags defaultPathFlags
+    , commandOptions = nixStyleOptions pathOptions
+    }
+
+-------------------------------------------------------------------------------
+-- Flags
+-------------------------------------------------------------------------------
+
+data PathOutputFormat
+  = JSON
+  | KeyValue
+  deriving (Eq, Ord, Show, Read, Enum, Bounded)
+
+data PathFlags = PathFlags
+  { pathCompiler :: Flag Bool
+  , pathOutputFormat :: Flag PathOutputFormat
+  , pathDirectories :: Flag [ConfigPath]
+  }
+  deriving (Eq, Show)
+
+defaultPathFlags :: PathFlags
+defaultPathFlags =
+  PathFlags
+    { pathCompiler = mempty
+    , pathOutputFormat = mempty
+    , pathDirectories = mempty
+    }
+
+pathOutputFormatParser :: ReadE (Flag PathOutputFormat)
+pathOutputFormatParser = ReadE $ \case
+  "json" -> Right $ Flag JSON
+  "key-value" -> Right $ Flag KeyValue
+  policy ->
+    Left $
+      "Cannot parse the status output format '"
+        <> policy
+        <> "'"
+
+pathOutputFormatString :: PathOutputFormat -> String
+pathOutputFormatString JSON = "json"
+pathOutputFormatString KeyValue = "key-value"
+
+pathOutputFormatPrinter
+  :: Flag PathOutputFormat -> [String]
+pathOutputFormatPrinter = \case
+  (Flag format) -> [pathOutputFormatString format]
+  NoFlag -> []
+
+pathOptions :: ShowOrParseArgs -> [OptionField PathFlags]
+pathOptions showOrParseArgs =
+  [ option
+      []
+      ["output-format"]
+      "Output format of the requested path locations"
+      pathOutputFormat
+      (\v flags -> flags{pathOutputFormat = v})
+      ( reqArg
+          (intercalate "|" $ map pathOutputFormatString [minBound .. maxBound])
+          pathOutputFormatParser
+          pathOutputFormatPrinter
+      )
+  , option
+      []
+      ["compiler-info"]
+      "Print information of the project compiler"
+      pathCompiler
+      (\v flags -> flags{pathCompiler = v})
+      (yesNoOpt showOrParseArgs)
+  ]
+    <> map pathOption [minBound .. maxBound]
+  where
+    pathOption s =
+      option
+        []
+        [pathName s]
+        ("Print cabal's " <> pathName s)
+        pathDirectories
+        (\v flags -> flags{pathDirectories = Flag $ concat (flagToList (pathDirectories flags) <> flagToList v)})
+        (noArg (Flag [s]))
+
+-- | A path that can be retrieved by the @cabal path@ command.
+data ConfigPath
+  = ConfigPathCacheDir
+  | ConfigPathLogsDir
+  | ConfigPathStoreDir
+  | ConfigPathConfigFile
+  | ConfigPathInstallDir
+  deriving (Eq, Ord, Show, Enum, Bounded)
+
+-- | The configuration name for this path.
+pathName :: ConfigPath -> String
+pathName ConfigPathCacheDir = "cache-dir"
```

Comment:
`--cache-dir` seems like a deprecated flag, should we rather use `remote-repo-cache` like the `~/.cabal/config` entry?

Reply:
I think I just extrapolated from the names of the other ConfigPaths. The purpose of the `path` command was not to explain what could be configured, but rather where `cabal` would write its files. (Of course there is no reason it shouldn't be able to do both.)

Reply:
Yeah, but what *is* the `cache-dir`? What was it supposed to query? It seems the name `cache-dir` was used internally a long time ago, and there is no mention of a `cache-dir` flag nowadays in any of the v2-commands. The documentation used to say, the flag name mirrors the config name, but afaict there is no such config entry?
Are you fine with renaming this flag or simply removing it?

Reply:
It is `$HOME/.cache/cabal` (depending on how your XDG variables are set). Used for logs and tarballs and such. I don't know if it can be configured outside setting either XDG or `CABAL_DIR`. It's pretty important to be able to find it, but I have no opinion on what it is called.

Reply:
Without `XDG` variables:
```
> cabal run exe:cabal -- path
Warning: this is a debug build of cabal-install with assertions enabled.
cache-dir: /home/hugin/.cabal/packages
logs-dir: /home/hugin/.cabal/logs
store-dir: /home/hugin/.cabal/store
config-file: /home/hugin/.cabal/config
installdir: /home/hugin/.cabal/bin
```

It contains package source tarballs and the index cache, but not the logs directory.

So, there is no direct translation for this unless the user is using `XDG` variables? Perhaps we should split this flag into all the different directories this is supposed to encompass?

Reply:
Sounds fine to me, but I don't have a particularly informed opinion.

Reply:
I think this flag should be named `remote-repo-cache`, as this is the name of the config, and what it should represent in both `XDG` and legacy behaviour. Also, that's the actual implementation.

Perhaps by `cache-dir`, you were referring to the top-level location `~/.cabal` and `XDG_CACHE_HOME/cabal`?

Reply:
> Perhaps by cache-dir, you were referring to the top-level location ~/.cabal and $XDG_CACHE_HOME/cabal?

Yes, that is what I would have expected it to mean.

Reply:
Ah, I understand now, thanks! Then I definitely will rename and add a flag that does actually does that.